### PR TITLE
[CI] dev-python/ipython < 7 cleanup

### PR DESCRIPTION
**Pull request by mgorny**

### CODE REVIEW

File: profiles/base/package.use.mask [hunk 1]
```diff
@@ -7,6 +7,23 @@
 # This file is only for generic masks. For arch-specific masks (i.e.
 # mask everywhere, unmask on arch/*) use arch/base.
 
+# Michał Górny <mgorny@gentoo.org> (2019-12-02)
+# Python 2.7-compatible packages optionally needing dev-python/ipython.
+# Note that all but [doc] flags here are optional runtime deps,
+# so no functionality is lost.
+app-misc/gourmet ipython
+dev-python/jira ipython jirashell
+dev-python/matplotlib doc
```
Comment (heroxbd) [inline profiles/base/package.use.mask]: Hi, I don't understand why masking is needed if python2_7 are moved from the dependents of ipython.  Could you please enlighten me?

Comment (mgorny) [reply profiles/base/package.use.mask]: I've presumed that of the two alternatives, temporarily disabling doc building would be less of a problem than killing Python 2 support entirely. Not to mention the latter would imply having to drop it from a whole new bunch of revdeps…

Comment (heroxbd) [reply profiles/base/package.use.mask]: OK, I understand now.  A further question taking matplotlib as example: How do you confirm only `dev-python/matplotlib[doc]` depends on` dev-python/ipython[python_targets_python2_7]` but not `dev-python/matplotlib` itself?  Are there some tools to aid this judgement?

Comment (mgorny) [reply profiles/base/package.use.mask]: I presume that whoever bumped it last time had verified the dependencies. Python world is not consistent enough to make it possible for good tooling to be created. It's handiwork if you want to check it.

**Comment by mgorny** [comment]
I'm going to work some more on eliminating masks in favor of cond-deps.

**Comment by heroxbd** [comment]
Thank you for your explanation.

Just out of curiosity, how bad is the idea of masking python_targets_python2_7 in the profile `use.mask`?  I think only less than 10 packages will be broken.

**Comment by mgorny** [comment]
More like hundreds that do not support py3 at all.

**Comment by heroxbd** [comment]
> More like hundreds that do not support py3 at all.

OK.  I had a wrong estimation.

**Comment by TheChymera** [comment]
This commit doesn't seem to be affecting the packages I maintain. Am I missing something?

**Comment by mgorny** [comment]
Maybe it's one of package.use.mask-ed.

**Pull request closed without merging**